evm/vm/forks/sharding/vmc_handler.py

```diff
@@ -44,27 +46,35 @@ class UnknownShard(Exception):
     pass
 
 
+@to_dict
+def deserialize_header_bytes(header_bytes):
+    header_sedes = CollationHeader.get_sedes()
+    # assume all fields are padded to 32 bytes
+    obj_size = 32
+    assert len(header_bytes) == obj_size * len(header_sedes)
+    for idx, field in enumerate(CollationHeader.fields):
+        field_name, field_type = field
+        start_index = idx * obj_size
+        field_bytes = header_bytes[start_index:(start_index + obj_size)]
+        if field_type == rlp.sedes.big_endian_int:
+            # remove the leading zeros, to avoid `not minimal length` error in deserialization
```

Comment:
is this a bug in the `rlp` library?  If so can you open an issue for it in that repository?

Reply:
I'm not sure. It just requires the `big_endian_int` to be in minimal length (e.g. `b'\x00\x01'` should be `b'\x01'`)
The error is like this
```shell
self = <rlp.sedes.big_endian_int.BigEndianInt object at 0x111d6bb38>
serial = b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x06'

    def deserialize(self, serial):
        if self.l is not None and len(serial) != self.l:
            raise DeserializationError('Invalid serialization (wrong size)',
                                       serial)
        if self.l is None and len(serial) > 0 and serial[0:1] == ascii_chr(0):
            raise DeserializationError('Invalid serialization (not minimal '
>                                      'length)', serial)
E           rlp.exceptions.DeserializationError: Invalid serialization (not minimal length)

/home/xxx/.pyenv/versions/py36env/lib/python3.6/site-packages/rlp/sedes/big_endian_int.py:40: DeserializationError
```
Will dig more into it to see if it is an issue.
Thank you:)